Homework2/Task1/Classes/ExtensionClass.cs

```diff
@@ -0,0 +1,54 @@
+﻿static class ExtensionClass
+{
+ public static DiagonalMatrix AddMatrixes(this DiagonalMatrix matrixOne, DiagonalMatrix matrixTwo)
+ {
+  bool isSizeEqual = matrixOne.Size == matrixTwo.Size ? true : false;
+  int sizeOfNewMatrix;
+  int[] elementsOfNewMatrix;
+
+
+  if (isSizeEqual)
+  {
+   elementsOfNewMatrix = new int[matrixOne.Size];
+
+   for (int i = 0; i < matrixOne.Size; i++)
+   {
+    elementsOfNewMatrix[i] = matrixOne[i, i] + matrixTwo[i, i];
+   }
+  }
+  else
+  {
+   if (matrixOne.Size > matrixTwo.Size)
```

Comment:
lines 22-33 and 37-48 contain repeatable code. 
Swap your matrixes if Two > One, so One is always biggest. 
Then sum them and set 0 when needed.